app/controllers/application_controller.rb

```diff
@@ -35,10 +36,33 @@ def after_sign_out_path_for(*)
     redirect_to "/bo/pages/permission"
   end
 
+  def authenticate_and_authorize_active_user
+    return if skip_auth_on_this_controller?
+
+    authenticate_user!
+    redirect_to "/bo/pages/deactivated" if current_user_is_deactivated?
+  end
+
   # http://jacopretorius.net/2014/01/force-page-to-reload-on-browser-back-in-rails.html
   def back_button_cache_buster
     response.headers["Cache-Control"] = "no-cache, no-store, max-age=0, must-revalidate"
     response.headers["Pragma"] = "no-cache"
     response.headers["Expires"] = "Fri, 01 Jan 1990 00:00:00 GMT"
   end
+
+  private
+
+  def skip_auth_on_this_controller?
+    # Don't authorize and authenticate pages from HighVoltage or Devise
+    # Normally we'd use a skip_before_action, but these controllers are in gems
+    controller = params[:controller]
+    controller.include?("pages") || controller.include?("devise")
+  end
+
+  def current_user_is_deactivated?
```

Comment:
I do find this method confusing. Can we change the name to: 
```
  def current_user_cannot_use_back_office?
```
Or, since this is used to redirect to a specific page that is `deactivated`, we can change the implementaation to be 
```
def current_user_is_deactivated?
  !(current_user && current_user.active?)
end
```
Actually by using the second approach, we can get rid of the extra code added in the Ability in regard to `can :use_back_office`